src/software/utils/main_fisheyeProjection.cpp

```diff
@@ -0,0 +1,431 @@
+// This file is part of the AliceVision project.
+// Copyright (c) 2017 AliceVision contributors.
+// Copyright (c) 2016 openMVG contributors.
+// This Source Code Form is subject to the terms of the Mozilla Public License,
+// v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include <aliceVision/numeric/numeric.hpp>
+#include <aliceVision/image/io.hpp>
+#include <aliceVision/image/Sampler.hpp>
+#include <aliceVision/image/convertion.hpp>
+#include <aliceVision/system/Logger.hpp>
+#include <aliceVision/system/cmdline.hpp>
+#include <aliceVision/camera/camera.hpp>
+
+#include <dependencies/vectorGraphics/svgDrawer.hpp>
+
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp> 
+
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/imagebuf.h>
+#include <OpenImageIO/imagebufalgo.h>
+
+#include <string>
+#include <iostream>
+#include <iterator>
+#include <fstream>
+#include <vector>
+#include <boost/regex.hpp>
+
+// These constants define the current software version.
+// They must be updated when the command line is changed.
+#define ALICEVISION_SOFTWARE_VERSION_MAJOR 1
+#define ALICEVISION_SOFTWARE_VERSION_MINOR 0
+
+using namespace aliceVision;
+
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+namespace oiio = OIIO;
+
+
+/**
+ * @brief Function to map 3D coordinates onto a 2D image according a spherical projection
+ */
+class SphericalMapping
+{
+public:
+
+  static Vec3 get3DPoint(const Vec2& pos2d, int width, int height)
+  {
+    const double x = pos2d(0) - width;
+    const double y = height/2.0 - pos2d(1);
+
+    const double longitude = M_PI * 2.0 * x / width;  // between -PI and PI
+    const double latitude = M_PI * y / height;        // between -PI/2 and PI/2
+
+    const double Px = cos(latitude) * cos(longitude);
+    const double Py = cos(latitude) * sin(longitude);
+    const double Pz = sin(latitude);
+
+    return Vec3(Px, Py, Pz);
+  }
+
+  static Vec2 get2DCoordinates(const Vec3& ray, const int inSize)
+  {
+    const double Px = ray(0);
+    const double Py = ray(1);
+    const double Pz = ray(2);
+
+//    float aperture = 2.0 * std::atan(36.0 / (8.0 * 2.0));
+//    float aperture = 2.0 * asin(1.0 / (2.0 * fNumber));
+
+    float r = 2.0 * atan2(sqrt(Square(Px) + Square(Pz)), Py) / M_PI;
```

Comment:
* 2.f
* all three are const
* maybe better add `std::`  to the functions atan sqrt etc 
* it would be better to use
```c++
#include <boost/math/constants/constants.hpp>

namespace bm::float =  boost::math::float_constants;

.... Py) / bm::float::pi
```
but I agree if it is too much.. ;-)